### ISSUE TITLE
How to interpret cntk results using HTKMLFWriter

### ISSUE THREAD

**Issue by vidyatshetty** [Microsoft/CNTK#2247]
Hi, I am using cntk for speaker recognition. So far I have completed data preparation using  HTK Toolkit. I used features and .mlf files in cntk. My configuration and output file file looks like this. I am wondering how to interpret results? Also If i use the HTKMLFWriter the output will not be wriiten in output file. Please help me solve these issues I am going through.

Config file

command=TimitTrainSimple:TimitTestSimple:TimitWriteSimple
# command=TimitWriteSimple

# deviceId=-1 for CPU, >=0 for GPU devices
DeviceNumber=0
#stderr=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Demo

precision=double

traceLevel = 0

#######################################
#  TRAINING CONFIG (Simple, Fixed LR) #
#######################################

TimitTrainSimple=[
    action=train

    modelPath=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Models/TrainSimple.dnn

    # deviceId=-1 for CPU, >=0 for GPU devices 
    deviceId=$DeviceNumber$

    traceLevel=2

    # Notation xxx:yyy*n:zzz is equivalent to xxx, then yyy repeated n times,
    #  then zzz
    # example: 10:20*3:5 is equivalent to 10:20:20:20:5
    SimpleNetworkBuilder=[
        layerSizes=39:50*3:2
        trainingCriterion=CrossEntropyWithSoftmax
        evalCriterion=ClassificationError
        layerTypes=Sigmoid
        initValueScale=1.0
        applyMeanVarNorm=true
        uniformInit=true
        needPrior=true
        ]
    SGD=[
        epochSize=0 
        #means epochSize is the size of the training set
        #epochSize=20480
        minibatchSize=256:1024
        learningRatesPerMB=0.8:3.2*14:0.08
        momentumPerMB=0.9
        dropoutRate=0.0
        maxEpochs=25		# Full experiment
        #maxEpochs=1		# Small experiment for debugging
           numMBsToShowResult = 1
    ]
       
    # Parameter values for the reader
    reader=[
      # reader to use
      readerType=HTKMLFReader

      #readMethod=blockRandomize
      readMethod=rollingWindow

      miniBatchMode=full
      verbosity=1
      randomize=auto
      frameMode=true
       
      features=[
	  dim=39
	  type=real
	  scpFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\scriptFile.scp
      ]
    

      labels=[
      type=category
	mlfFileList=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\mlfList.txt
	mlfFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\scriptFile2.mlf.txt
	labelMappingFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\TimitStateList.txt
	labelDim=2
	labelType=category
      ]
    ]
]
 

#######################################
#  TESTING CONFIG (Simple)            #
#######################################

TimitTestSimple=[
    action=test

    modelPath=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Models\TrainSimple.dnn

    # deviceId=-1 for CPU, >=0 for GPU devices 
    deviceId=$DeviceNumber$

    traceLevel=2

	# epochSize=0 means epochSize is the size of the training set
        epochSize=0
        minibatchSize=256:1024
       
    # Parameter values for the reader
    reader=[
      # reader to use
      readerType=HTKMLFReader

      readMethod=rollingWindow

      miniBatchMode=Partial
      randomize=Auto
      verbosity=1   

	#numMBsToShowResult=10
	# evalNodeNames

      features=[
	  dim=39
	  type=real
	  scpFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\testing.txt
      ]

      labels=[
      type=category
      mlfFileList=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\mlfTestList.txt
	mlfFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\testingMlf.txt
	labelMappingFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\TimitStateList.txt
	labelDim=2
	labelType=category
      ]
    ]

]

#######################################
#  OUTPUT RESUTLS (Simple)            #
#######################################

TimitWriteSimple=[
    action=write

    modelPath=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Models/TrainSimple.dnn

    # deviceId=-1 for CPU, >=0 for GPU devices 
    deviceId=$DeviceNumber$

    outputNodeNames=ScaledLogLikelihood
    traceLevel=2
 

    # Parameter values for the reader
    reader=[
      # reader to use
      readerType=HTKMLFReader

      features=[
	  dim=39
	  type=real
	  scpFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Data\testing.txt
      ]

    ]
 # writer=[
  #   wrecords=50000 #Number of records to allocate space for in file
  #   writerType=HTKMLFWriter
  #   ScaledLogLikelihood = [
  #       dim = 2
  #scpFile=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Output\CntkTimitOutput.scp
  #       type=Real			# Must be uppercase Real?
#	]
	
    
  outputPath = "C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Output\TestProbs.txt"
]
 dump weight and bias values
TIMIT_DumpNodes = [
  action = "dumpNode"
  modelPath=C:\Users\vidya\CNTK-2-1-Windows-64bit-GPU-1bit-SGD\cntk\speakerRecognition-Vidya\Models\TrainSimple.dnn
  printValues = "true"
]
]

Output file
0.866145 0.724829
0.866163 0.724823
0.866177 0.724828
0.865997 0.724919
0.866128 0.724856
0.866099 0.724870
0.866146 0.724825
0.866116 0.724847
0.866025 0.724914
0.866022 0.724915
0.866034 0.724900
0.866105 0.724847
0.866230 0.724756
0.866141 0.724823
0.866108 0.724849
0.866079 0.724860
0.866141 0.724842
0.866121 0.724874
0.866127 0.724875
0.866067 0.724891
0.866035 0.724896

**Comment by cha-zhang** [comment]
The output is the output of the linear layer before softmax. If you need output that sum to 1, you need to apply softmax yourself.

**Comment by vidyatshetty** [comment]
Thank you writing.
Also if i include HTKMLFWriter nothing will be written into output file. Could you please let me know what is the reason.
